### ISSUE TITLE
Error with resize_token_embeddings and LLama model

### ISSUE THREAD

**Issue by lccnl** [huggingface/transformers#25729]
### System Info

- `transformers` version: 4.32.0
- Platform: Linux-5.15.0-1038-gcp-x86_64-with-glibc2.31
- Python version: 3.10.7
- Huggingface_hub version: 0.16.4
- Safetensors version: 0.3.2
- Accelerate version: 0.21.0
- Accelerate config:    not found
- PyTorch version (GPU?): 2.0.1+cu117 (True)

### Who can help?

@ArthurZucker 

### Information

- [ ] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

In the last version of the method `resize_token_embeddings`, the `pad_token_to_multiple=N` arg allows to increase the dimension to the first  int divisibile by N larger than `new_num_tokens`. 

However, in the method, the `config.vocab_size` is still updated to the initial [new_num_tokens](https://github.com/huggingface/transformers/blob/main/src/transformers/modeling_utils.py#L1438).

For the LLama model, this leads to an error when the logits are flattened [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/llama/modeling_llama.py#L837). 

### Expected behavior

We should update the vocab_size to the new_num_tokens that accounts for the arg pad_token_to_multiple

**Comment by ydshieh** [comment]
cc @SunMarc Maybe ?

**Comment by ArthurZucker** [comment]
Non that's actually me, I'll open a PR to fix this
